### PR TITLE
refactored homebrew role to be command: module free

### PR DESCRIPTION
**Pull request by ninjabong**
also renamed defaults -> vars so the output of name: {{ var }} is nicer during a playbook run.

### CODE REVIEW

**Comment by ninjabong** [comment]
I left it using ~/src/github.com/mxcl for now per our discussion... easy enough to change and somewhere in my xc-custom's history is bits to detect and migrate from mxcl to Homebrew if we decide to do that later.

**Comment by rricard** [comment]
I definitely like the brew_install_path option !

File: roles/homebrew/tasks/main.yml [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
+  sudo: True
+  synchronize: src={{ brew_install_path }}/ dest={{ brew_repo_path }} copy_links=yes
+  when: brew_installed.stat.exists == false
 
-- name: clean up /usr/local
+- name: fix user acl on {{ brew_repo_path }} and clean up {{ brew_install_path }}
   sudo: True
-  command: rm -rf /usr/local
-  when: brew_installed|failed
+  file: path={{ item.path }} state={{ item.state|default('directory') }}
```
Comment (rricard) [inline roles/homebrew/tasks/main.yml:19]: Wow ! cool much better ! Really like it !

File: roles/homebrew/tasks/main.yml [hunk 2]
```diff
@@ -1,31 +1,34 @@
...
+    - { path: '{{ brew_repo_path }}/bin' }
+    - { path: '{{ brew_install_path }}' , state: 'absent' }
+  when: brew_installed.stat.exists == false
 
-- name: link the repo to the brew installation path
+- name: ensure {{ brew_repo_path }} is linked to the {{ brew_install_path }}
   sudo: True
   file: state=link src={{ brew_repo_path }} dest={{ brew_install_path }} owner={{ ansible_user_id }}
-  when: brew_installed|failed
+  when: brew_installed.stat.exists == false
```
Comment (rricard) [inline roles/homebrew/tasks/main.yml:30]: Ok, in the end we'll try to get rid of that !

**Comment by rricard** [comment]
I merge right away ! And I'll try to bring it to the detached brew repo !

**Comment by rricard** [comment]
Oh ! btw, now you can directly use the origin repo to create your branches+prs.

**Comment by ninjabong** [comment]
Yeah I just more comfortable using private branch for now, much safer 8)

**Pull request merged**